src/org/javarosa/core/services/transport/payload/ByteArrayPayload.java

```diff
@@ -126,4 +124,15 @@ public int getTransportId() {
         //TODO: Most messages can include this data
         return -1;
     }
+
+    @Override
+    public String toString() {
+        try {
+            return new String(payload, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
```

Comment:
Perhaps we should write a message via the logger.

Reply:
Yes, was thinking of that. Thanks @dcbriccetti Was looking at the `Logger` class but realized it's depreciated. Which logger should I use? Looking in other parts of the code for dets